onnxruntime/test/shared_lib/utils.cc

```diff
@@ -5,15 +5,31 @@
 
 #ifdef USE_CUDA
 #include <cuda_runtime.h>
-void cuda_add(int64_t, float*, const float*, const float*);
+template <typename T1, typename T2, typename T3>
+void cuda_add(int64_t, T3*, const T1*, const T2*);
 #endif
 
 void MyCustomKernel::Compute(OrtKernelContext* context) {
+  // The kernel compute's logic handles the following cases:
+  // 1) First input: float, Second input: float, Output: float
```

Comment:
Even though the custom op says the inputs are all dynamic (i.e.) can be any type - we limit the implementation to just these as they are enough for illustrative purposes.